src/fcntl.rs

```diff
@@ -504,3 +504,16 @@ mod posix_fadvise {
         Errno::result(res)
     }
 }
+
+#[cfg(any(
+    target_os = "linux",
+    target_os = "android",
+    target_os = "emscripten",
+    target_os = "fuchsia",
+    any(target_os = "wasi", target_env = "wasi"),
+    target_env = "freebsd"
+))]
+pub fn posix_fallocate(fd: RawFd, offset: libc::off_t, len: libc::off_t) -> Result<libc::c_int> {
```

Comment:
A better return type would be `Result<()>`, because the only non-error value ever returned is 0.

Reply:
The spec says that `posix_fallocate` returns the error number and it does not set `errno`.
`fallocate` returns `-1` on error and set `errno`. The two functions are quite different.

Reply:
I have converted to return type to `Result<()>`. However, I have concerns on whether we should follow POSIX and return Errno directly?